learningequality/kolibri
# Add Auto-saving and Search to Lesson Resource Selection

### Summary

1. Adds a search bar to Lesson Resource Selection
1. Removes 'Save' button and automatically saves resource changes when selecting resources
1. Adds a select-all button to topics
1. Updates copy on UI and different snackbar notifications
1. Updates some of the copy on the Create Exam Page
1. Fixes whitespace in .feature files

**Search**
![search](https://user-images.githubusercontent.com/10248067/45639303-7bb8ba80-ba64-11e8-8157-122ec3a787ad.gif)


**Autosave**
![autosave](https://user-images.githubusercontent.com/10248067/45639310-7eb3ab00-ba64-11e8-81c1-4d41127cc804.gif)

**Create Exam Page**

![screen shot 2018-09-17 at 3 02 58 pm](https://user-images.githubusercontent.com/10248067/45652884-f268ae80-ba8a-11e8-864f-c9e601e1cd54.png)

### Reviewer guidance

**Testing search**

1. Search term with results
1. Search term with no results
1. Search from within UI
1. Search using URL (put search term at the very of the URL)
1. Exiting search page should go back to previous page or channels listing

**Testing auto save**

1. Test toggling select-all
1. Toggling single object
1. Resources are kept when going from topic to topic or preview or search
1. Resources are kept when exiting the selection workflow

### References


…

----

### Contributor Checklist

- [ x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')

### Reviewer Checklist
